### DISCUSSION TITLE
Dynamic Waypoints

### DISCUSSION THREAD

**Discussion by willishf** [iNavFlight/inav#8759] [Ideas]
Working through the various options in INAV to accomplish the ability to have dynamic GPS coordinate(s) as a waypoint(s) for a drone mission. 

A use case would be to respond to an alarm via motion detection, gate etc on a large property. Each alarm would have a known GPS coordinate and if the alarm is triggered send the GPS coordinate to the drone which may be on the ground or in flight as a remote camera. 

I can see how it would be done if the drone is on the ground as a typical mission. More interested in how to do it dynamically when the drone is in the air. Would use MSP to send the GPS coordinate programmatically and have it go to that location.   Then once it arrives or before it gets to that location if you send another GPS coordinate it becomes the new primary target. 

Any thoughts or suggestions on how to do that with existing iNAV FC code?

**Comment by stronnag** [comment]
`GCS NAV` aka "follow me" (canonical answer to such questions).

**Reply by willishf** [reply]
Thanks will look into that option.

**Comment by stronnag** [comment]
Start with the [wiki](https://github.com/iNavFlight/inav/wiki/INAV-Remote-Management,-Control-and-Telemetry#follow-me-gcs-nav
)
